internal/consensus/state.go

```diff
@@ -1310,8 +1311,21 @@ func (cs *State) createProposalBlock(ctx context.Context) (*types.Block, error)
 		lastExtCommit = &types.ExtendedCommit{}
 
 	case cs.LastCommit.HasTwoThirdsMajority():
-		// Make the commit from LastCommit
-		lastExtCommit = cs.LastCommit.MakeExtendedCommit(cs.state.ConsensusParams.Feature)
+		// Make the commit from LastCommit.
+		//
+		// Note we can't aggregate a commit when vote extensions are enabled
+		// because votes are different.
+		_, blsKey := cs.privValidatorPubKey.(*bls12381.PubKey)
+		canBeAggregated := blsKey &&
```

Comment:
This also means that blsKeys cannot be used in a non-aggregated manner? Is that desired?

Reply:
Yes! There's no point in using BLS keys unless you want sig aggregation. Sig aggregation is faster, and the resulting commit is a magnitude smaller in size.